coz-shop/src/components/ProductsList.js

```diff
@@ -1,12 +1,35 @@
+import { useEffect, useState } from "react"
 import Menu from "./menu"
 import Product from "./Product"
-const ProductsList = ({menuOn,menuHandle,ProductArr}) =>{
+import Category from "./Category"
+import Toast from "./Toast"
+const ProductsList = ({menuOn,menuHandle,ProductArr,bookmark,bookmarkHandle,toastEvent,filterResult}) =>{
+   const [focus , setFocus] = useState('All')
+   const [curruntArr , setCurruntArr] = useState(ProductArr)
+   const addMessege = '북마크에 추가되었습니다.'
+   const delMessege = '북마크가 제거되었습니다.'
+   const focusHandle = (e) =>{
+      setFocus(e)
+   }
+
+   useEffect(()=>{
+      if(focus==='All'){
+         setCurruntArr(ProductArr)
+      }else{
+         setCurruntArr(
+            ProductArr.filter((el)=>{
+            return el.type === focus
+         }))
+      }
+   },[focus])
    return <div className="page_Container">
+      {toastEvent ? <Toast messege={filterResult? delMessege : addMessege} markOn={filterResult? false : true}/> : ''}
```

Comment:
```suggestion
      {toastEvent ? <Toast messege={filterResult? delMessege : addMessege} markOn={filterResult? false : true}/> : null}
```

빈 문자열보다는 null이 좀 더 좋습니다. (빈 문자열은 HTML에 무의미한 text가 생길 수 있음)